code/datums/components/jousting.dm

```diff
@@ -1,27 +1,77 @@
+ */
 /datum/component/jousting
+	///The current person holding parent.
+	var/mob/living/current_holder
+	///The current direction of the jousting.
 	var/current_direction = NONE
-	var/max_tile_charge = 5
-	var/min_tile_charge = 2 //tiles before this code gets into effect.
+	///How many tiles we've charged up thus far
 	var/current_tile_charge = 0
-	var/movement_reset_tolerance = 3 //deciseconds
-	var/unmounted_damage_boost_per_tile = 0
-	var/unmounted_knockdown_chance_per_tile = 0
-	var/unmounted_knockdown_time = 0
-	var/mounted_damage_boost_per_tile = 2
-	var/mounted_knockdown_chance_per_tile = 20
-	var/mounted_knockdown_time = 20
-	var/requires_mob_riding = TRUE //whether this only works if the attacker is riding a mob, rather than anything they can buckle to.
-	var/requires_mount = TRUE //kinda defeats the point of jousting if you're not mounted but whatever.
-	var/mob/current_holder
-	var/current_timerid
-
-/datum/component/jousting/Initialize()
+
+	///How much of an increase in damage is achieved every tile moved during jousting.
+	var/damage_boost_per_tile
+	///The boosted chances of a knockdown occuring while jousting.
+	var/knockdown_chance_per_tile
+	///How much of an increase in knockdown is achieved every tile moved during jousting, if it knocks down.
+	var/knockdown_time
+	///The max amount of tiles before you can joust someone.
+	var/max_tile_charge
+	///The min amount of tiles before you can joust someone.
+	var/min_tile_charge
+
+/datum/component/jousting/Initialize(
+	damage_boost_per_tile = 2,
+	knockdown_chance_per_tile = 20,
+	knockdown_time = 2 SECONDS,
+	max_tile_charge = 5,
+	min_tile_charge = 2,
+)
 	if(!isitem(parent))
 		return COMPONENT_INCOMPATIBLE
+	src.damage_boost_per_tile = damage_boost_per_tile
+	src.knockdown_chance_per_tile = knockdown_chance_per_tile
+	src.knockdown_time = knockdown_time
+	src.max_tile_charge = max_tile_charge
+	src.min_tile_charge = min_tile_charge
+
+	RegisterSignal(parent, COMSIG_ATOM_EXAMINE, PROC_REF(on_examine))
 	RegisterSignal(parent, COMSIG_ITEM_EQUIPPED, PROC_REF(on_equip))
 	RegisterSignal(parent, COMSIG_ITEM_DROPPED, PROC_REF(on_drop))
 	RegisterSignal(parent, COMSIG_ITEM_ATTACK, PROC_REF(on_attack))
+	RegisterSignal(parent, COMSIG_TRANSFORMING_ON_TRANSFORM, PROC_REF(on_transform))
 
+/datum/component/jousting/UnregisterFromParent()
+	. = ..()
+	UnregisterSignal(parent, list(
+		COMSIG_ATOM_EXAMINE,
+		COMSIG_ITEM_EQUIPPED,
+		COMSIG_ITEM_DROPPED,
+		COMSIG_ITEM_ATTACK,
+		COMSIG_TRANSFORMING_ON_TRANSFORM,
+	))
+
+/datum/component/jousting/proc/on_examine(datum/source, mob/user, list/examine_list)
+	SIGNAL_HANDLER
+	examine_list += span_notice("It can be used on a vehicle for jousting, dealing potential knockdowns and additional damage.")
+
+/datum/component/jousting/proc/on_transform(obj/item/source, mob/user, active)
+	SIGNAL_HANDLER
+	if(!user)
+		return
+
+	if(active)
+		INVOKE_ASYNC(src, PROC_REF(on_equip), user)
+	else
+		INVOKE_ASYNC(src, PROC_REF(on_equip), user)
```

Comment:
These are both doing the same thing? Is one supposed to be `on_drop`?

Reply:
...yes